server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserService.kt

```diff
@@ -45,16 +45,35 @@ class PublicUserService(@Autowired private val publicUserRepository: PublicUserR
                 losses = 0,
                 ties = 0,
                 score = 0.0,
-                tier = LeaderBoardEnum.TIER_PRACTICE,
+                tier = TierTypeDto.TIER_PRACTICE,
                 isDailyChallengeCompleted = false,
                 tutorialLevel = 1,
             )
         publicUserRepository.save(publicUser)
     }
 
+    fun updateLeaderboard(publicUsers: List<PublicUserEntity>) {
+        for (i in publicUsers.indices) {
+            val user = publicUserRepository.findById(publicUsers[i].userId).get()
+            if (i < topPlayers) {
+                publicUserRepository.save(user.copy(tier = TierTypeDto.TIER1))
+            } else {
+                publicUserRepository.save(user.copy(tier = TierTypeDto.TIER2))
+            }
+        }
+    }
+
+    fun updateTempLeaderboardByTier() {
+        updateLeaderboard(publicUserRepository.findAll())
+    }
+
+    fun updateLeaderboard() {
+        updateLeaderboard(publicUserRepository.findAll(Sort.by(Sort.Order.desc("rating"))))
+    }
+
     fun getLeaderboard(page: Int?, size: Int?, tier: TierTypeDto?): List<LeaderboardEntryDto> {
-        val pageRequest = PageRequest.of(page ?: 0, size ?: 10, Sort.by(Sort.Direction.DESC, "rating"))
-        return publicUserRepository.findAll(pageRequest).content.map {
+        val pageRequest = PageRequest.of(page ?: 0, size ?: 10, Sort.by(Sort.Order.desc("rating")))
+        return publicUserRepository.findAll(pageRequest).content.filter { it.tier == tier }.map {
```

Comment:
- Ig here u can make use of `methods` provided by `MongoRepository` to filter players by their Tier